### PR TITLE
Fix client_reporter traceback.

### PR DESCRIPTION
**Pull request by oliver-sanders**
The delete at line 146 was causing traceback due to the key being removed by the delete at line 133.

### CODE REVIEW

File: lib/cylc/network/https/client_reporter.py [hunk 1]
```diff
@@ -131,19 +131,22 @@ def signout(self, server_obj):
...
+        except KeyError:
             # Already forgotten.
             pass
         self._housekeep()
 
     def _housekeep(self):
         """Forget inactive clients."""
 
-        for uuid in self.clients.keys():
+        for uuid in self.clients.copy():
```
Comment (hjoliver) [inline lib/cylc/network/https/client_reporter.py]: why this change?

Comment (oliver-sanders) [inline lib/cylc/network/https/client_reporter.py]: Have fixed so that this makes sense.
An entry from the dictionary is being deleted whilst this loop is in progress, making a copy ensures that the contents don't change during the loop.

Comment (hjoliver) [inline lib/cylc/network/https/client_reporter.py]: But doesn't `dict.keys()` return a list, so deletion of list elements does not affect the original source dict?

Comment (hjoliver) [inline lib/cylc/network/https/client_reporter.py]: Sorry - let me rephrase that!

Comment (hjoliver) [inline lib/cylc/network/https/client_reporter.py]: But doesn't dict.keys() return a list, so deletion of dict elements does not affect the list over which we're iterating?

Comment (oliver-sanders) [inline lib/cylc/network/https/client_reporter.py]: In python 2 yes, in python 3 `dict.keys()` returns a view. I've changed this to `dict.items()` as my initial commit didn't fix the immediate problem.

Comment (hjoliver) [inline lib/cylc/network/https/client_reporter.py]: OK fine, we're anticipating Python 3...
